### PR TITLE
[Feature] Add new respiration rate methods

### PR DESCRIPTION
**Pull request by Tam-Pham**

### CODE REVIEW

File: neurokit2/rsp/rsp_rate.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    rsp = signal_resample(rsp_cleaned, sampling_rate=sampling_rate, desired_sampling_rate=desired_sampling_rate)
+
+    # Define paramters
+    window_length = int(desired_sampling_rate * window)
+
+    rsp_rate = []
+    for start in np.arange(0, N, hop_size):
+        window_segment = rsp[start: start + window_length]
+        if len(window_segment) < window_length:
+            continue # the last frames that are smaller than windlow_length
```
Comment (JanCBrammer) [inline neurokit2/rsp/rsp_rate.py:80]: maybe `break` instead of `continue` (no need to iterate over the rest)?

File: neurokit2/rsp/rsp_rate.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    >>> rsp_rate = nk.rsp_rate(rsp_cleaned, method="peaks")
+    >>> rsp_rate2 = nk.rsp_rate(rsp_cleaned, method="xcorr")
+    """
+
+    if method.lower() in ["peak", "peaks", "signal_rate"]:
+        if peaks is None:
+            peaks, info = rsp_peaks(rsp_cleaned, sampling_rate=sampling_rate, method=peak_method)
+        rsp_rate = signal_rate(peaks, sampling_rate=sampling_rate, desired_length=len(rsp_cleaned))
+
+    if method.lower() in ["cross-correlation", "xcorr"]:
```
Comment (DominiqueMakowski) [inline neurokit2/rsp/rsp_rate.py:53]: I'd also probably do an if, elif, else error "wrong method" flow to throw informative errors

File: neurokit2/rsp/rsp_rate.py [hunk 3]
```diff
@@ -38,22 +35,28 @@ def rsp_rate(rsp_cleaned, peaks=None, sampling_rate=1000, window=10, hop_size=1,
...
-    >>> rsp_rate2 = nk.rsp_rate(rsp_cleaned, method="xcorr")
+    >>> rsp_rate_peak = nk.rsp_rate(rsp_cleaned, sampling_rate=sampling_rate, method="peaks")
+    >>> rsp_rate_xcorr = nk.rsp_rate(rsp_cleaned, sampling_rate=sampling_rate, method="xcorr")
     """
 
     if method.lower() in ["peak", "peaks", "signal_rate"]:
         if peaks is None:
             peaks, info = rsp_peaks(rsp_cleaned, sampling_rate=sampling_rate, method=peak_method)
-        rsp_rate = signal_rate(peaks, sampling_rate=sampling_rate, desired_length=len(rsp_cleaned))
+        rsp_rate = signal_rate(peaks, sampling_rate=sampling_rate, desired_length=len(rsp_cleaned), interpolation_method="linear")
```
Comment (DominiqueMakowski) [inline neurokit2/rsp/rsp_rate.py:48]: I wouldn't hardcode the interpolation_method nor put linear as a default, I think we discussed it in https://github.com/neuropsychology/NeuroKit/issues/243 and opted for monotonic if I remember correctly

Comment (Tam-Pham) [reply neurokit2/rsp/rsp_rate.py:48]: Yes, you are right. I added in for comparison and didn't remove. I will add interpolation method as an option

**Pull request merged**